integration/terraform/ec2/linux/main.tf

```diff
@@ -1,18 +1,58 @@
-resource "aws_instance" "integration-test" {
-  ami                    = data.aws_ami.latest.id
-  instance_type          = var.ec2_instance_type
-  key_name               = var.key_name
-  iam_instance_profile   = var.iam_instance_profile
-  vpc_security_group_ids = var.vpc_security_group_ids
+#####################################################################
+# Ensure there is unique testing_id for each test
+#####################################################################
+resource "random_id" "testing_id" {
+  byte_length = 8
+}
+
+#####################################################################
+# Generate EC2 Key Pair for log in access to EC2
+#####################################################################
+
+resource "tls_private_key" "ssh_key" {
+  count     = var.ssh_key == "" ? 1 : 0
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "aws_key_pair" "aws_ssh_key" {
+  count      = var.ssh_key == "" ? 1 : 0
+  key_name   = "ec2-key-pair-${random_id.testing_id.hex}"
+  public_key = tls_private_key.ssh_key[0].public_key_openssh
+}
+
+locals {
+  ssh_key_name        = var.ssh_key != "" ? var.key_name : aws_key_pair.aws_ssh_key[0].key_name
+  private_key_content = var.ssh_key != "" ? var.ssh_key : tls_private_key.ssh_key[0].private_key_pem
+}
```

Comment:
Is this a necessary part of the change

Reply:
IMO, not entirely. However, this help me reduces the necessary resources to create for when using integration test.

Reply:
We discussed separately about why the ssh key needs to be separate. We should revert this change

Reply:
I think its happen at GitHub Action level. If you decided not to use it locally, then its bring no harm. Moreover, if you decided not to use `ssh key` at Github Action, that should be fine too.